### PR TITLE
Payment MFE: Update toggle to support all product types

### PR DESCRIPTION
**Pull request by nasthagiri**
https://openedx.atlassian.net/browse/ARCH-1056

### CODE REVIEW

File: ecommerce/extensions/basket/views.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def get(self, request):
             except AlreadyPlacedOrderException:
                 return render(request, 'edx/error.html', {'error': _('You have already purchased these products')})
 
-            self._redirect_to_microfrontend_if_needed(request, products)
+            _redirect_to_payment_microfrontend_if_configured(request)
```
Comment (robrap) [inline ecommerce/extensions/basket/views.py:104]: I find a couple of things about this confusing.
1. I don't like the use of an exception for the common successful case, and
2. The inconsistency between how the Payment MFE and Basket case (get it, basket case) are handled adds to the confusion.
I think I'd prefer either `_redirect_to_payment_mfe_or_basket_page(request)` here, or `_get_payment_mfe_or_basket_page_url(request)` here.  The latter would leave a single call to `add_utm_params_to_url` here, which I think would be nice, so that would be my preference.
UDPATE: Or, you could move the single call of `add_utm_params_to_url` inside `_get_payment_mfe_or_basket_page_url(request)`, and then it would be less likely that they could get dropped in certain situations.  (Not impossible, but less likely.)

File: ecommerce/extensions/basket/views.py [hunk 2]
```diff
@@ -470,7 +461,7 @@ def get(self, request, *args, **kwargs):
         try:
             self.fire_segment_events(request, basket)
             self.verify_enterprise_needs(basket)
-            self._redirect_to_microfrontend_if_needed(request, basket)
+            _redirect_to_payment_microfrontend_if_configured(request)
```
Comment (robrap) [inline ecommerce/extensions/basket/views.py:464]: Again, I think it is odd to use RedirectException for a non-exceptional case.  I don't like the idea of a try/except where there is no successful try.

File: ecommerce/settings/_debug_toolbar.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # DJANGO DEBUG TOOLBAR CONFIGURATION
 DEBUG_TOOLBAR_CONFIG = {
-    'SHOW_TOOLBAR_CALLBACK': (lambda __: True),
+    'SHOW_TOOLBAR_CALLBACK': (lambda __: False),
```
Comment (robrap) [inline ecommerce/settings/_debug_toolbar.py]: Not sure if you intentionally added this.  I think we should consider it, but it should probably be its own PR with comments/README updated.  It is really the basket page specifically that has the most issues with the toolbar.

Comment (nasthagiri) [reply ecommerce/settings/_debug_toolbar.py]: No. I didn't intend to add this.  Removed.

File: ecommerce/extensions/basket/views.py (hunk 2 above)
Comment (nasthagiri) [reply ecommerce/extensions/basket/views.py:464]: I believe this is a personal coding style.  I prefer using exceptions even for non-exceptional cases since exceptions can be used to simplify code and redirect code-paths in a simpler way.

File: ecommerce/extensions/basket/views.py (hunk 1 above)
Comment (nasthagiri) [reply ecommerce/extensions/basket/views.py:104]: I'm fine with the way it is now.  The current code makes it clear that there is a shared common helper (`_redirect_to_payment_microfrontend_if_configured`) that is used by both Payment and Voucher views.  And the name of the helper indicates that the code-path is interruped with a redirect, if needed.

This is also throw-away temporary code so I'd prefer not to over-optimize.

Comment (robrap) [reply ecommerce/extensions/basket/views.py:104]: I really found the code confusing for the reasons I listed.
I think either:
1. we get a third opinion from another person who was not the author of the code, or 
2. you can just merge it as-is if you really think it is more clear than my recommendation, or
3. I'd be happy to refactor to make it more clear.
Feel free to go with whatever option you think is best.  Just let me know.  Thanks.

**Review by robrap — APPROVED** [review]
Let me know if you decide to merge as-is and whether or not you are ok with a follow-up (see comment).  Thanks.

File: ecommerce/extensions/basket/views.py (hunk 1 above)
Comment (nasthagiri) [reply ecommerce/extensions/basket/views.py:104]: I went ahead and merged.  I don't see the value in spending any more time on this.  If anyone chooses to refactor this further in a subsequent PR, they should go ahead.
